### PR TITLE
Add support for SQS

### PR DESCRIPTION
**Pull request by nickcaballero**
This PR adds support for SQS. Part of it involved signing the `Host` header, which is required by AWS.

### CODE REVIEW

File: src/internal/sqs.ts [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+}
+
+export interface ListQueuesCommand {
+    maxResults?: number
+    nextToken?: string
+    queueNamePrefix?: string
+}
+
+export interface ListQueuesResponse {
+    queues: string[]
```
Comment (oleiade) [inline src/internal/sqs.ts:130]:
```suggestion
     /**
      * A list of queue URLs, up to 1,000 entries, or the value of MaxResults you sent in the request.
      */
    queueUrls: string[]
```

File: src/internal/sqs.ts [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+    messageBody: string
+}
+
+export interface SendMessageResponse {
+    messageId: string
+    md5OfMessageBody: string
+}
+
+export interface ListQueuesCommand {
+    maxResults?: number
```
Comment (oleiade) [inline src/internal/sqs.ts:124]:
```suggestion
    /**
     * Maximum number of results to include in the response. Value range is 1 to 1000. 
     */
    maxResults?: number
```

File: src/internal/sqs.ts [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+    nextToken?: string
```
Comment (oleiade) [inline src/internal/sqs.ts:125]:
```suggestion
    /**
     * Pagination token to request the next set of results.
     */
    nextToken?: string
```

File: src/internal/sqs.ts [hunk 4]
```diff
@@ -0,0 +1,132 @@
...
+    queueNamePrefix?: string
```
Comment (oleiade) [inline src/internal/sqs.ts:126]:
```suggestion
    /**
     * A string to use for filtering the list results. Only those queues whose name begins with the specified string are returned.
     * Queue URLs and names are case-sensitive.
     */
    queueNamePrefix?: string
```

File: src/internal/sqs.ts [hunk 5]
```diff
@@ -0,0 +1,132 @@
...
+    }
+}
+
+export interface SendMessageCommand {
+    queueUrl: string
+    messageBody: string
+}
+
+export interface SendMessageResponse {
+    messageId: string
```
Comment (oleiade) [inline src/internal/sqs.ts:119]:
```suggestion
    /**
     * An attribute containing the MessageId of the message sent to the queue.
     */
    messageId: string
```

File: src/internal/sqs.ts [hunk 6]
```diff
@@ -0,0 +1,132 @@
...
+    md5OfMessageBody: string
```
Comment (oleiade) [inline src/internal/sqs.ts:120]:
```suggestion
    /**
     * An MD5 digest of the non-URL-encoded message body string. You can use this attribute to verify that Amazon SQS received the message correctly. Amazon SQS URL-decodes the message before creating the MD5 digest. 
     */
    md5OfMessageBody: string
```

**Review by oleiade — CHANGES_REQUESTED** [review]
Hey @nickcaballero 👋🏻 

Thanks a lot for your contribution! This is awesome, really great work 🎉 🦖 💯 

I've left a couple of comments as requested changes. Those are primarily about consistency in naming and documenting the code (we use those comments to write the official documentation later).

Before going forward with this, I would also like to ask you to add a README section dedicated to SQS in the same fashion as with the other services. On top of that, I would kindly ask you to add an example file in the `examples` folder. 

Once those items are addressed, we should be good to go moving forward with merging this 👏🏻 

PS: sorry for the rather late reply, Christmas happened 🎄 🎅🏻 ❄️

File: src/internal/signature.ts [hunk 7]
```diff
@@ -93,13 +93,12 @@ export class SignatureV4 {
...
         const scope = `${shortDate}/${region}/${service}/${constants.KEY_TYPE_IDENTIFIER}`
 
-        // FIXME: test wants us to leave host alone, but I'm unsure at this point
         // Required by the specification:
         //   "For HTTP/1.1 requests, you must include the host header at a minimum.
         //   Standard headers like content-type are optional.
         //   For HTTP/2 requests, you must include the :authority header instead of
         //   the host header. Different services might require other headers."
-        // request.headers[constants.HOST_HEADER] = request.hostname
+        request.headers[constants.HOST_HEADER] = request.hostname
```
[the remark this answers is not in the document]
Comment (oleiade) [reply src/internal/signature.ts:101]: This, indeed introduces a bug, and shall be reverted 👍🏻 See #29

**Comment by oleiade** [comment]
Looks good @nickcaballero 👍🏻 🙇🏻 
Thanks for integrating the requested changes, much appreciated!

Would you say the PR is ready to be merged ❓

**Comment by nickcaballero** [comment]
> Looks good @nickcaballero 👍🏻 🙇🏻 Thanks for integrating the requested changes, much appreciated!
> 
> Would you say the PR is ready to be merged ❓

Yeah I think so.

**Comment by oleiade** [comment]
Hi @nickcaballero, great 🤝 

Could you please rebase the current state of the `main` branch? I'll make sure to merge this as soon as it's done. Please let me know when it is and if I can assist you in any ways 😄

**Comment by nickcaballero** [comment]
@oleiade Done!

**Comment by oleiade** [comment]
Perfect, thank you very much 🙇🏻 

Merging, and this will go in our next release who doesn't have a set release date at this point, but I'll make sure to notify you here 👏🏻

**Pull request merged**